File: Samples/Digital signatures/SignWithAzureKeyVault/C#/SignWithAzureKeyVault.cs
using System;
using System.Diagnostics;
using System.Security.Cryptography.X509Certificates;
using Azure.Identity;
using Azure.Security.KeyVault.Keys;
using Azure.Security.KeyVault.Keys.Cryptography;

namespace BitMiracle.Docotic.Pdf.Samples
{
    public static class SignWithAzureKeyVault
    {
        public static void Main()
        {
            // IMPORTANT:
            // Change these variables using your vault url, key name, signing algorithm, and Azure credentials.
            const string VaultUrl = "https://YOUR-VAULT.vault.azure.net/";
            const string KeyName = "YOUR-KEY";
            var signingAlgorithm = SignatureAlgorithm.RS512;
            var credentials = new AzureCliCredential();

            var keyClient = new KeyClient(vaultUri: new Uri(VaultUrl), credential: credentials);
            KeyVaultKey key = keyClient.GetKey(KeyName);
            var cryptoClient = new CryptographyClient(key.Id, credentials);
            var signer = new AzureSigner(cryptoClient, signingAlgorithm);

            // This sample automatically generates self-signed certificate for the provided Azure key.
            // In real applications, you might want to use a CA signed certificate.
            using X509Certificate2 cert = AzureCertificateGenerator.Generate(key, cryptoClient, signingAlgorithm);

            // NOTE:
            // When used in trial mode, the library imposes some restrictions.
            // Please visit https://bitmiracle.com/pdf-library/trial-restrictions
            // for more information.

            string outputFileName = "SignWithAzureKeyVault.pdf";
            using (var pdf = new PdfDocument())
            {
                PdfPage page = pdf.Pages[0];
                PdfSignatureField field = page.AddSignatureField(50, 50, 200, 200);

                var options = new PdfSigningOptions(signer, new[] { cert })
                {
                    DigestAlgorithm = signer.DigestAlgorithm,
                    Field = field,
                };

                pdf.SignAndSave(options, outputFileName);
            }

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}");

            Process.Start(new ProcessStartInfo(outputFileName) { UseShellExecute = true });
        }
    }
}

File: Samples/Digital signatures/SignWithAzureKeyVault/VB.NET/SignWithAzureKeyVault.vb
Imports Azure.Identity
Imports Azure.Security.KeyVault.Keys
Imports Azure.Security.KeyVault.Keys.Cryptography
Imports System.Security.Cryptography.X509Certificates
Imports BitMiracle.Docotic.Pdf
Imports BitMiracle.Docotic

Namespace BitMiracle.Docotic.Pdf.Samples
    Public NotInheritable Class SignWithAzureKeyVault
        Public Shared Sub Main()
            ' IMPORTANT:
            ' Change these variables using your vault url, key name, signing algorithm, and Azure credentials.
            Const VaultUrl As String = "https://YOUR-VAULT.vault.azure.net/"
            Const KeyName As String = "YOUR-KEY"
            Dim signingAlgorithm = SignatureAlgorithm.RS512
            Dim credentials = New AzureCliCredential()

            Dim keyClient = New KeyClient(vaultUri:=New Uri(VaultUrl), credential:=credentials)
            Dim key As KeyVaultKey = keyClient.GetKey(KeyName)
            Dim cryptoClient = New CryptographyClient(key.Id, credentials)
            Dim signer = New AzureSigner(cryptoClient, signingAlgorithm)

            ' NOTE:
            ' Without a license, the library won't allow you to create or read PDF documents.
            ' To get a free time-limited license key, use the form on
            ' https://bitmiracle.com/pdf-library/download

            LicenseManager.AddLicenseData("PUT-LICENSE-HERE")

            Dim outputFileName As String = "SignWithAzureKeyVault.pdf"
            Using pdf = New PdfDocument()
                Dim page As PdfPage = pdf.Pages(0)
                Dim field As PdfSignatureField = page.AddSignatureField(50, 50, 200, 200)

                ' This sample automatically generates self-signed certificate for the provided Azure key.
                ' In real applications, you might want to use a CA signed certificate.
                Using cert As X509Certificate2 = AzureCertificateGenerator.Generate(key, cryptoClient, signingAlgorithm)
                    Dim options = New PdfSigningOptions(signer, {cert}) With {
                        .DigestAlgorithm = signer.DigestAlgorithm,
                        .Field = field
                    }

                    pdf.SignAndSave(options, outputFileName)
                End Using
            End Using

            Console.WriteLine($"The output is located in {Environment.CurrentDirectory}")
            Process.Start(New ProcessStartInfo(outputFileName) With {
                .UseShellExecute = True
            })
        End Sub
    End Class
End Namespace